src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -371,4 +380,21 @@ case class DeltaSource(
       case other => throw new UnsupportedOperationException(s"Unknown ReadLimit: $other")
     }
   }
+
+  /** Extracts whether users provided the option to time travel a relation. */
+  private def getStartingVersion: Option[Long] = {
```

Comment:
nit: can we either cache the result of this as a lazy val or just make this method a lazy val? The computation may be a bit expensive